### DISCUSSION TITLE
How to enable forecast?

### DISCUSSION THREAD

**Discussion by summitbri** [briis/hass-weatherflow2mqtt#102] [Q&A]
This is probably a really obvious one.  The doc mentions 'If you have enabled the Forecast option, then there is a possibility to create a Weather Entity' but I don't see an option to do that.  All of the sensors are reporting perfectly in HA, but no forecast and I fear I haven't turned it on :)

Any help would be great, thanks!

EDIT- figured it out, I was using the wrong # for the station (the number in the default station name isn't the right one.)

**Comment by briis** [comment]
Glad you figured it out, but re-reading the docs I can also see that it is not clear enough how to enable the Forecast, so I will update that.

**Reply by alvinchen1** [reply]
Hi, so looks like I am stupid.  Awesome integration BTW.  I also read the doc, but don't understand where to enable the Forecast option.  Is it in the Addon / Docker configuration (I see Forecast Interval which I have set) or is it somewhere else?  I went through the Tempest app and the website (tempestwx.com) looking at every possible setting for where to enable Forecast, but I can't seem to find it, and I don't see Sensors like 

sensor.hub_hb_[my number]_weather
sensor.tempest_[my number]_temperature

but all my other sensors are working.  Am I missing something really obvious?

**Reply by briis** [reply]
As you are using the Add-On version, 
- go to -> *Configuration* -> *Add-Ons*
- Then click *WeatherFlow to MQTT*
- Now click *Configuration* in the top Menu
- Once here, toggle the Switch *Show unused optional configuration options*.
- Now you should see two fields you need to fill in: *STATION_ID* and *STATION_TOKEN*. Once these are filled in with your personal data, click *Save* below the section, and you will be asked if you want to restart the Add-On. Say yes, and shortly after that, you should get the Weather Entities.

**Reply by alvinchen1** [reply]
Awesome.  That worked.  

Just one mistake I made, and perhaps the documentation could be updated a bit to be slightly clearer.

Based on https://github.com/briis/hass-weatherflow2mqtt#docker-environment-variables
> STATION_ID: Enter your Station ID for your WeatherFlow Station. Default value is blank.

The STATION_ID I put in originally was the ST-000XXXXX.  When I looked through the WeatherFlow to MQTT logs, I saw an error 

> INFO:weatherflow2mqtt.weatherflow_mqtt:Setting up Hub sensor: Weather
> ERROR:weatherflow2mqtt.forecast:Your API Key is invalid or does not support this operation
> WARNING:weatherflow2mqtt.forecast:Forecast Server was unresponsive. Skipping forecast update

The correct STATION_ID is the five digit number that for example when you are on https://tempestwx.com/station/XXXXX/

Putting that in and everything works!  I think for a newbie like me, it was a little confusing whether the STATION_ID needed in configuration is ST-000XXXXX or the last part of the URL https://tempestwx.com/station/XXXXX/.

**Reply by briis** [reply]
Fair point. We will look at how to improve the documentation. Thanks and glad you are up and running
